engines/cryo/cryo.cpp

```diff
@@ -0,0 +1,116 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "common/scummsys.h"
+
+#include "common/config-manager.h"
+#include "common/debug.h"
+#include "common/debug-channels.h"
+#include "common/error.h"
+#include "graphics/surface.h"
+#include "graphics/screen.h"
+#include "graphics/palette.h"
+#include "common/system.h"
+
+#include "engines/util.h"
+
+#include "cryo/cryo.h"
+#include "cryo/eden.h"
+
+namespace Cryo {
+
+CryoEngine *g_ed = nullptr;
+
+CryoEngine::CryoEngine(OSystem *syst, const ADGameDescription *gameDesc) : Engine(syst), _gameDescription(gameDesc) {
+	// Put your engine in a sane state, but do nothing big yet;
+	// in particular, do not load data from files; rather, if you
+	// need to do such things, do them from run().
+
+	// Do not initialize graphics here
+	// Do not initialize audio devices here
+
+	// However this is the place to specify all default directories
+//	const Common::FSNode gameDataDir(ConfMan.get("path"));
+//	SearchMan.addSubDirectoryMatching(gameDataDir, "sound");
+
+	// Here is the right place to set up the engine specific debug channels
+	DebugMan.addDebugChannel(kCryoDebugExample, "example", "this is just an example for a engine specific debug channel");
+	DebugMan.addDebugChannel(kCryoDebugExample2, "example2", "also an example");
+
+	// Don't forget to register your random source
+	_rnd = new Common::RandomSource("cryo");
+	_debugger = nullptr;
+
+	_game = nullptr;
+	_video = nullptr;
+	_screenView = nullptr;
+
+	_showHotspots = false;
+
+	g_ed = this;
+}
+
+CryoEngine::~CryoEngine() {
+	debug("CryoEngine::~CryoEngine");
+
+	// Dispose your resources here
+	delete _rnd;
+	delete _game;
+	delete _video;
+	delete _screenView;
+	delete _debugger;
+
+	// Remove all of our debug levels here
+	DebugMan.clearAllDebugChannels();
+}
+
+Common::Error CryoEngine::run() {
+	_game = new EdenGame(this);
+	_video = new HnmPlayer(this);
+	_screenView = new View(320, 200);
+	_debugger = new Debugger(this);
+
+	///// CLTimer
+	_timerTicks = 0;   // incremented in realtime
+
+	// Initialize graphics using following:
+	initGraphics(320, 200, false);
+	_screen.create(320, 200, Graphics::PixelFormat::createFormatCLUT8());
+
+	// Additional setup.
+	debug("CryoEngine::init");
+
+	// Your main even loop should be (invoked from) here.
+	debug("CryoEngine::go: Hello, World!");
+
+	// This test will show up if -d1 and --debugflags=example are specified on the commandline
+	debugC(1, kCryoDebugExample, "Example debug call");
+
+	// This test will show up if --debugflags=example or --debugflags=example2 or both of them and -d3 are specified on the commandline
+	debugC(3, kCryoDebugExample | kCryoDebugExample2, "Example debug call two");
```

Comment:
These debug statements are meant to be placeholders, and look to be leftovers... they should be removed, eventually